File: 1. Two Sum/main.go
// https://leetcode.com/problems/two-sum/?envType=problem-list-v2&envId=rabvlt31
package mai

func twoSum(nums []int, target int) []int {
	numsMapping := make(map[int]int)

	for index, num := range nums {
		requiredVal := target - num

		if prevIndex, ok := numsMapping[requiredVal]; ok {
			return []int{prevIndex, index}
		}
		numsMapping[num] = index

	}
	return []int{}
}

File: 1. Two Sum/main.js
//https://leetcode.com/problems/two-sum/description/?envType=problem-list-v2&envId=rabvlt31
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let numsMapping = new Map()

    for(let i =0; i < nums.length;i++) {
        let requiredVal = target - nums[i]
         if (numsMapping.has(requiredVal)) {
            return [numsMapping.get(requiredVal), i];
        }
        numsMapping.set(nums[i], i)
    }
    return [];
};